lib/core/managementClient.ts

```diff
@@ -532,6 +555,175 @@ export class ManagementClient extends LinkEntity {
     }
   }
 
+
+  /**
+   * Get all the rules on the Subscription.
+   * @returns Promise<RuleDescription> A list of rules.
+   */
+  async getRules(): Promise<RuleDescription[]> {
+    try {
+      const request: AmqpMessage = {
+        body: {
+          top: types.wrap_int(100),
```

Comment:
@amarzavery Adding a max int value here like the way .Net does, results in the code hanging with the below every 15 seconds. Where is this limitation at? We could ask the user to provide the count, but I was curious to know if you knew anything else about this

![image](https://user-images.githubusercontent.com/16890566/49196649-8b942b00-f340-11e8-9662-115369a53d51.png)